isso/db/__init__.py

```diff
@@ -114,10 +126,72 @@ def first(rv):
                         ids.extend(rv)
                         flattened[id].update(set(rv))
 
-                for id in flattened.keys():
-                    for n in flattened[id]:
-                        con.execute(
-                            "UPDATE comments SET parent=? WHERE id=?", (id, n))
+                con.execute("BEGIN TRANSACTION")
+                try:
+                    for id in flattened.keys():
+                        for n in flattened[id]:
+                            con.execute(
+                                "UPDATE comments SET parent=? WHERE id=?", (id, n))
+
+                    con.execute('PRAGMA user_version = 3')
+                    con.execute("COMMIT")
+                    logger.info("Database migration to version 3 finished, %i rows changed", con.total_changes)
+                except sqlite3.Error as e:
+                    logger.error("Database migration to version 3 failed: %s", e)
+                    con.execute("ROLLBACK")
+
+        # add "notification" field to "comments" (moved from Comments class to migration)
+        if self.version == 3:
+            with sqlite3.connect(self.path) as con:
+                self.migrate_to_version_4(con)
```

Comment:
Is there a particular reason you decided to factor out only this migration step into its own function? Afaics you're only saving one indentation level.

Reply:
I thought it would be more readable that way. In order not to create additional nesting due to checking that the `notification` field already exists.

Reply:
There is too much nesting, I agree.